pkg/java/java.go

```diff
@@ -114,7 +114,7 @@ func (o *Opts) checkVersion() error {
 	if err != nil {
 		return err
 	}
-	if !o.VersionConstraints.Check(currentVersion) {
+	if o.VersionConstraints != nil && !o.VersionConstraints.Check(currentVersion) {
```

Comment:
this is fine but that value could be also set by config in some `Configure` method